tests/default_tasks_test/jumanji_envs_test.py

```diff
@@ -0,0 +1,131 @@
+"""Test default rastrigin using MAP Elites"""
+
+import functools
+
+import jax
+import jax.numpy as jnp
+import jumanji
+import numpy as np
+import pytest
+
+from qdax.core.neuroevolution.buffers.buffer import QDTransition
+from qdax.core.neuroevolution.networks.networks import MLP
+from qdax.tasks.jumanji_envs import (
+    jumanji_scoring_function,
+    make_policy_network_play_step_fn,
+)
+from qdax.types import Descriptor, Observation
+
+
+def test_jumanji_utils() -> None:
+
+    seed = 0
+    policy_hidden_layer_sizes = (128, 128)
+    episode_length = 200
+    population_size = 100
+    batch_size = population_size
+
+    # Instantiate a Jumanji environment using the registry
+    env = jumanji.make("Snake-6x6-v0")
+
+    # Reset your (jit-able) environment
+    key = jax.random.PRNGKey(0)
+    state, _timestep = jax.jit(env.reset)(key)
+
+    # Interact with the (jit-able) environment
+    action = env.action_spec().generate_value()  # Action selection (dummy value here)
+    state, _timestep = jax.jit(env.step)(state, action)
+
+    # Init a random key
+    random_key = jax.random.PRNGKey(seed)
+
+    # get number of actions
+    num_actions = env.action_spec().maximum + 1
+
+    policy_layer_sizes = policy_hidden_layer_sizes + (num_actions,)
+    policy_network = MLP(
+        layer_sizes=policy_layer_sizes,
+        kernel_init=jax.nn.initializers.lecun_uniform(),
+        final_activation=jax.nn.softmax,
+    )
+
+    def observation_processing(observation: jumanji.types.Observation) -> Observation:
+        network_input = jnp.ravel(observation)
+        return network_input
+
+    play_step_fn = make_policy_network_play_step_fn(
+        env=env,
+        policy_network=policy_network,
+        observation_processing=observation_processing,
+    )
+
+    # Init population of controllers
+    random_key, subkey = jax.random.split(random_key)
+    keys = jax.random.split(subkey, num=batch_size)
+
+    # compute observation size from observation spec
+    observation_size = np.prod(np.array(env.observation_spec().shape))
+
+    fake_batch = jnp.zeros(shape=(batch_size, observation_size))
+    init_variables = jax.vmap(policy_network.init)(keys, fake_batch)
+
+    # Create the initial environment states
+    random_key, subkey = jax.random.split(random_key)
+    keys = jnp.repeat(jnp.expand_dims(subkey, axis=0), repeats=batch_size, axis=0)
+    reset_fn = jax.jit(jax.vmap(env.reset))
+
+    init_states, init_timesteps = reset_fn(keys)
+
+    # Prepare the scoring function
+    def bd_extraction(
+        data: QDTransition, mask: jnp.ndarray, linear_projection: jnp.ndarray
+    ) -> Descriptor:
+        """Compute feet contact time proportion.
+
+        This function suppose that state descriptor is the feet contact, as it
+        just computes the mean of the state descriptors given.
+        """
```

Comment:
Is this description adequate?
First of all, I don't know what are the _feet_ for snakes (and I don't find the work "feet" in the [environment code](https://github.com/instadeepai/jumanji/blob/main/jumanji/environments/games/snake/env.py)).

Also, at first sight, it seems it is taking the mean of all the observations vector (which is fine), but not only the mean of the feet contacts. But I may have missed something!

Reply:
It's a left over of a copy/paste haha. Not relevant at all here. Will be removed.